### PR TITLE
Update trmm and rocBLAS status code handling to use rocBLAS 3 functionality

### PR DESCRIPTION
**Pull request by msimberg**
Uses the out-of-place `rocblas_trmm` whenever available. Defines `ROCBLAS_V3` to make the new version of the function available. Still uses the old `rocblas_trmm_outofplace` for older versions. I've only tested compilation on clariden; running still seems to be problematic.

Also adds a few `rocblas_status` enumerations that were added in rocBLAS 3.1.0 (ROCm 5.7.0 onwards). I've also updated the rocSOLVER `getErrorString` function to simply forward to the rocBLAS `getErrorString` as they should do exactly the same (and were out-of-sync with each other).

### CODE REVIEW

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
The test failure is a hang in the hermitian multiplication test again (same as https://github.com/eth-cscs/DLA-Future/issues/603#issuecomment-1851956301, so not adding it again). I've restarted the job.

File: include/dlaf/gpu/rocblas/error.h [hunk 1]
```diff
@@ -21,6 +21,10 @@ namespace dlaf::gpublas::rocblas::internal {
 inline std::string getErrorString(rocblas_status st) {
   // clang-format off
   switch (st) {
+#if ROCBLAS_VERSION_MAJOR >= 3 && (ROCBLAS_VERSION_MINOR >= 1 || ROCBLAS_VERSION_MAJOR >= 4)
```
Comment (rasolca) [inline include/dlaf/gpu/rocblas/error.h:24]: I find the following more readable:
```suggestion
#if (ROCBLAS_VERSION_MAJOR == 3 && (ROCBLAS_VERSION_MINOR >= 1) || ROCBLAS_VERSION_MAJOR >= 4
```

Comment (msimberg) [reply include/dlaf/gpu/rocblas/error.h]: Agree, that looks much better, thanks! Updated in https://github.com/eth-cscs/DLA-Future/pull/1091/commits/a9da88766269ef0cd8d07816e8fe9150a19c5949.

**Review by rasolca — APPROVED** [review]

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
There's definitely something funky going on with HEMM: https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/6128997608#L2341. Already reported on #603, but it might be worth bumping that up the priority list. I'll see if I can reproduce it myself in the next few weeks.

**Review by albestro — APPROVED** [review]

**Comment by rasolca** [comment]
cscs-ci run

**Pull request merged**